flow/layers/display_list_layer.cc

```diff
@@ -132,6 +133,29 @@ void DisplayListLayer::Paint(PaintContext& context) const {
     }
   }
 
+  if (context.enable_leaf_layer_tracing) {
+    const auto canvas_size = context.leaf_nodes_canvas->getBaseLayerSize();
```

Comment:
Will this be prohibitively expensive to do on a series of frames?

For example, what if instead of eagerly storing the rasterized leaf, we instead stored the leaf with enough information to rasterize it later.

Then, we could _always_ record the time it takes to rasterize the whole layer tree. We could update the snapshot store with those times. _Later_, a user could request rasterizations of the top N frames with their broken down raster times (and the amount of time it actually took to rasterize the whole scene).

WDYT?

Reply:
I think this approach would also let us keep DRM on when this is enabled, but maybe I'm misunderstanding.